### PR TITLE
[TypeDeclaration] Skip ArrayShapeFromConstantArrayReturnRector on key has colon

### PR DESCRIPTION
**Pull request by samsonasik**
Closes https://github.com/rectorphp/rector-src/pull/1961

### CODE REVIEW

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

File: rules-tests/TypeDeclaration/Rector/ClassMethod/ArrayShapeFromConstantArrayReturnRector/Fixture/skip_key_with_colon_multi.php.inc [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?php
+
+namespace Rector\Tests\TypeDeclaration\Rector\ClassMethod\ArrayShapeFromConstantArrayReturnRector\Fixture;
+
+final class SkipKeyWithColonMulti
+{
+    public function run(string $name)
+    {
+        return ['prefix:key' => $name, 'another:key' => $name];
```
Comment (TomasVotruba) [inline rules-tests/TypeDeclaration/Rector/ClassMethod/ArrayShapeFromConstantArrayReturnRector/Fixture/skip_key_with_colon_multi.php.inc:9]: I've noticed also these fails:

```php
return ['{key}' => $name];
```

**Comment by TomasVotruba** [comment]
Thank you :+1:

File: rules-tests/TypeDeclaration/Rector/ClassMethod/ArrayShapeFromConstantArrayReturnRector/Fixture/skip_key_with_colon_multi.php.inc (hunk 1 above)
Comment (samsonasik) [reply rules-tests/TypeDeclaration/Rector/ClassMethod/ArrayShapeFromConstantArrayReturnRector/Fixture/skip_key_with_colon_multi.php.inc:9]: I created PR https://github.com/rectorphp/rector-src/pull/1968 for it.

**Pull request merged**